### ISSUE TITLE
Agent merging / marking as bad duplicate

### ISSUE THREAD

**Issue by AJLinn** [ArctosDB/arctos#1568]
I'm curious if folks are taking the time to look at the agent activity reports of possible duplicate agents before they mark them as bad duplicates or if they're just doing it marked on similar names?

Occasionally I miss the opportunity to go in and check when the notice comes through that one has been marked (there was a flurry recently) and I don't realize it until after it's been merged. This happened today:

> Agent merger for Jack Sullivan --> Jack M. Sullivan is complete.

These two men are clearly not the same person.

> Jack M. Sullivan
Department of Biological Sciences
Life Sciences South Room 252
University of Idaho
Moscow, ID 83844-3051

Not the same as the Jack Sullivan who collected this rifle and "operated the University of Alaska power plant for many years": http://arctos.database.museum/guid/UAM:EH:1087-0001

This can be determined with a few questions of the folks who are about to be affected by an agent merge. I just had to re-create Jack Sullivan's agent profile. Hopefully I hadn't entered any additional information in it other than his name - I'm in the process of adding biographical information to these records and if it gets merged/deleted I'm not going to enjoy having to re-create that info.

Perhaps we can add a line or two in our handbook about updating agents - that before you mark as a bad duplicate that you make the effort to check that potential bad duplicate's activity and confirm that it's most likely to be the same. If you can't determine that, contact the collection contact for those other collections and ask. You could be saving someone else a great deal of time in having to re-create agent profiles.

thank you - that is all.

**Comment by campmlc** [comment]
I agree, and would add that before any merger of bad duplicates, the agent
activity should be checked for any relationships or activity in other
collections. I also like the idea of having some minimum information
required as to who created the agent name and when, at the very least in
remarks ("MSB Mammals bulkloaded agent name"). Then we could contact the
relevant collection manager prior to merging as duplicate.

**Comment by dustymc** [comment]
That's probably my fault - SORRY!

I do try to look at everything, but obviously don't dig deep enough or just miss something sometimes.

Relationships actively prevent mergers; remarks occasionally (very occasionally I think/hope!) lead us here.

Who and when is in the agent record and doesn't really need replicated in remarks. 


```
 AgentID 21314007 created by Angela Linn on 2018-06-13 09:43:43.0 
```


Something about the agent ("collects butterflies in Alaska") is sometimes useful, but can also be misleading - lots of folks collect various taxa, donate items to cultural collections, etc.

I'm very open to better models for merging agents. I think flagging as duplicates IS asking everyone involved in the agent. We could not merge until we hear back from everyone or something, but I suspect that would lead to agents never really being merged. (Nonresponsiveness lead to this "delete unless stopped" model.) I flag a lot of these, usually looking around when someone force-creates an agent - I could just stop doing that as I'm seldom close enough to the collection to really KNOW anyway.

If we do change something, http://handbook.arctosdb.org/documentation/agent.html#deletingmerging-agents should be updated.

>Agents lacking evidence to the contrary should be marked as duplicates

**Comment by AJLinn** [comment]
>  I think flagging as duplicates IS asking everyone involved in the agent. 

I agree, the current system usually works fine for me - I think this is the first time I missed one. It seems like sometimes the bad duplicates come thru in waves and I usually check them if I get the message.

> Something about the agent ("collects butterflies in Alaska") is sometimes useful, but can also be misleading - lots of folks collect various taxa, donate items to cultural collections, etc.

I agree, BUT I always add "UAM Ethnology & History" to the remarks of every agent I create. If I find an agent that I share with another collection, I add this line to the remarks. I do this because I think it's a valuable piece of information to be able to share with anyone who is interested - how many people have contributed objects to our collection (though I see now that there is a "used by collection" dropdown I can select as well - any way we can get a number tally when that comes up?).

**Comment by dustymc** [comment]
>number tally

New Issue please - shouldn't be a problem, but we'd need to work out details.

You can also Reports/WriteSQL

```
select 
  agent.preferred_agent_name 
from
  agent,
  collector,
  cataloged_item,
  collection
where
  agent.agent_id=collector.agent_id and
  collector.collection_object_id=cataloged_item.collection_object_id and
  cataloged_item.collection_id=collection.collection_id and
  guid_prefix='UAM:EH'
group by 
  preferred_agent_name
order by 
  preferred_agent_name
```

2714 rows selected.

And one of these is probably wrong...


S. J. Saario Furs
S. T. Saario

**Comment by Jegelewicz** [comment]
I am closing this as duplicate of 2319. Also, code is in sql_cheats - https://github.com/ArctosDB/documentation-wiki/blob/gh-pages/_sql_cheats/collection_agent_search.markdown
